realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -675,6 +687,11 @@ private Scanner getFullStringScanner(InputStream in) {
     public <E extends RealmModel> E createObject(Class<E> clazz) {
         checkIfValid();
         Table table = schema.getTable(clazz);
+        // Check and throw the exception earlier for a better exception message.
+        if (table.hasPrimaryKey()) {
+            throw new RealmException(String.format("'%s' has a primary key, use" +
+                    " 'createObject(Class<E>, Object)' to create it.", clazz.getSimpleName()));
```

Comment:
`clazz.getSimpleName()` returns obfuscated name if ProGuard was applied.
`Table.tableNameToClassName(table.getName())` is better I think.